shell/common/engine_unittests.cc

```diff
@@ -299,4 +302,44 @@ TEST_F(EngineTest, PassesLoadDartDeferredLibraryErrorToRuntime) {
   });
 }
 
+TEST_F(EngineTest, NotifyIdleMoreThanOncePerFiveSeconds) {
+  PostUITaskSync([this] {
+    MockRuntimeDelegate client;
+    auto mock_runtime_controller =
+        std::make_unique<MockRuntimeController>(client, task_runners_);
+
+    EXPECT_CALL(*mock_runtime_controller, NotifyIdle(200, 100)).Times(1);
+    EXPECT_CALL(*mock_runtime_controller, NotifyIdle(400, 0)).Times(1);
```

Comment:
So this line here is verifying that we didn't get called (the second param is 0) at the 13s mark.

Reply:
Ah yea, a bit weird to follow since mocking and calling are in different spaces and it isn't straitforward.  You might want to add a comment like:
`Here we assert NotifyIdle is called with the zero hint since we haven't advanced 5 (or name of timeout constant) seconds yet.`

Reply:
I can do that - unfortunately it's this way because we std::move the pointer after this, and I didn't want to grab out a copy of the unique_ptr.